plugins/plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/MachineProviderImplTest.java

```diff
@@ -1574,29 +1527,28 @@ public void shouldAddEnvVarsFromMachineConfigToContainerOnDevInstanceCreationFro
         envVarsFromConfig.put("ENV_VAR1", "123");
         envVarsFromConfig.put("ENV_VAR2", "234");
 
-        provider = new ComposeMachineProviderImpl(dockerConnector,
-                                                  dockerConnectorConfiguration,
-                                                  credentialsReader,
-                                                  dockerMachineFactory,
-                                                  dockerInstanceStopDetector,
-                                                  containerNameGenerator,
-                                                  Collections.emptySet(),
-                                                  Collections.emptySet(),
-                                                  Collections.emptySet(),
-                                                  Collections.emptySet(),
-                                                  null,
-                                                  workspaceFolderPathProvider,
-                                                  PROJECT_FOLDER_PATH,
-                                                  false,
-                                                  false,
-                                                  Collections.emptySet(),
-                                                  Collections.emptySet(),
-                                                  SNAPSHOT_USE_REGISTRY,
-                                                  MEMORY_SWAP_MULTIPLIER,
-                                                  emptySet());
+        provider = new MachineProviderImpl(dockerConnector,
```

Comment:
I see the code like this 
```java
new MachineProviderImpl(dockerConnector,
                        dockerConnectorConfiguration,
                        credentialsReader,
                        dockerMachineFactory,
                        dockerInstanceStopDetector,
                        Collections.emptySet(),
                        Collections.emptySet(),
                        Collections.emptySet(),
                        Collections.emptySet(),
                        null,
                        workspaceFolderPathProvider,
                        PROJECT_FOLDER_PATH,
                        false,
                        false,
                        Collections.emptySet(),
                        Collections.emptySet(),
                        SNAPSHOT_USE_REGISTRY,
                        MEMORY_SWAP_MULTIPLIER,
                        emptySet());
```
in all the tests. It's logical that there is some difference between them but still it is not obvious from object 
creation, i understand that test name may help on this consideration but still i think it is not really nice to have such code.

I suggest to create a helpful class which may produce the needed instance of `MachineProviderImpl` e.g.
`MachineProviderImplBuilder` with a method which may produce default instance of it and than set only the difference. 
The example:

```java
MachineProviderImplBuilder.initDefault()
                          .setMemorySwapMultiplier(...)
                          .build()
```
In this way you may simplify the code for the readers who are not really into constructor parameters